iothub/device/src/CommandRequest.cs

```diff
@@ -13,12 +13,12 @@ public sealed class CommandRequest
         private readonly byte[] _data;
         private readonly PayloadConvention _payloadConvention;
 
-        internal CommandRequest(string commandName, string componentName = default, byte[] data = default, PayloadConvention payloadConvention = default)
+        internal CommandRequest(string commandName, PayloadConvention payloadConvention, string componentName = default, byte[] data = default)
```

Comment:
I am being really nit picky: I kind of like the command name and component name next to each other; what do you think about moving the payload convention to be beginning?

Reply:
I'm good with that, I just wanted it to be a required param.